simulator-core/src/test/java/com/evbox/everon/ocpp/simulator/it/ChangeEvseAvailabilityIntegrationTest.java

```diff
@@ -0,0 +1,61 @@
+package com.evbox.everon.ocpp.simulator.it;
+
+import com.evbox.everon.ocpp.simulator.mock.StationSimulatorSetUp;
+import com.evbox.everon.ocpp.simulator.station.Station;
+import com.evbox.everon.ocpp.simulator.station.evse.EvseStatus;
+import com.evbox.everon.ocpp.v20.message.station.ChangeAvailabilityRequest;
+import org.junit.jupiter.api.Test;
+
+import static com.evbox.everon.ocpp.simulator.mock.ExpectedCount.twice;
+import static com.evbox.everon.ocpp.simulator.mock.ExpectedRequests.bootNotificationRequest;
+import static com.evbox.everon.ocpp.simulator.mock.ExpectedRequests.statusNotificationRequest;
+import static com.evbox.everon.ocpp.simulator.mock.ExpectedResponses.bootNotificationResponse;
+import static com.evbox.everon.ocpp.simulator.mock.ExpectedResponses.statusNotificationResponse;
+import static com.evbox.everon.ocpp.simulator.support.JsonMessageTypeFactory.createCall;
+import static com.evbox.everon.ocpp.simulator.support.StationConstants.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+public class ChangeEvseAvailabilityIntegrationTest extends StationSimulatorSetUp {
+
+    @Test
+    void shouldChangeEvseStatusToUnavailable() {
+
+        ocppMockServer
+                .when(bootNotificationRequest())
+                .thenReturn(bootNotificationResponse());
+
+        ocppMockServer
+                .when(statusNotificationRequest(), twice())
+                .thenReturn(statusNotificationResponse());
+
+        stationSimulatorRunner.run();
+
+        ocppMockServer.waitUntilConnected();
+
+        ocppServerClient.findSender(STATION_ID).sendMessage(createChangeAvailabilityRequest());
+
+        await().untilAsserted(() -> {
+            Station station = stationSimulatorRunner.getStation(STATION_ID);
+
+            assertThat(station.getState().findEvse(DEFAULT_EVSE_ID).getEvseStatus()).isEqualTo(EvseStatus.UNAVAILABLE);
+
+            ocppMockServer.verify();
```

Comment:
I believe verify method doesn't make much sense. If you mock some behavior which is required for condition to happen - it will fail before verify is called. E.g. transaction has to start if OCPP server replies with successful AuthorizeResponse

Reply:
Nope. It may not fail. This is the main purpose of the verify function. You have to verify that all requests have been received and all expectations are met. As a reference https://docs.spring.io/autorepo/docs/spring-framework/4.2.0.RC2/spring-framework-reference/html/integration-testing.html

For example: if we make a bug in the code which prevents sending the request, verify function will catch it.
As as reference: https://docs.spring.io/autorepo/docs/spring-framework/4.2.0.RC2/spring-framework-reference/html/integration-testing.html

Reply:
By calling it here you only verify if boot notification and status notification were sent. This is already covered by dedicated tests. If assertion would depend on server response - it would fail before verify is called.

Reply:
Yes, this is what I'm mocking and verifying. 
Verify and assertion is 2 different things. Thats why good mock-libraries have assert-function and verify-function.
Assertion might not fail. In this case status might change, but the code might not send StatusNotification request.